src/Components/programs/Program.js

```diff
@@ -1,34 +1,58 @@
 import React, { useState } from 'react';
 import { StyleSheet, TouchableOpacity, View, Text, FlatList } from 'react-native';
+import { connect } from 'react-redux';
 
+import { completeTraining } from '../../Store/Actions/programAction';
 import theme from '../../styles/theme.style';
 
-const Program = props => {
+export const Program = props => {
   const { id, title, trainings } = props.program;
-  const firstTodoTrainingIndex = 0; // Faking it for now
 
-  const onPress = item => props.navigation.navigate('TrainingPage', { training: trainings[firstTodoTrainingIndex] });
-  const width = `${(firstTodoTrainingIndex * 100) / trainings.length}%`;
+  const programCompleteTrainings = props.completeTrainings
```

Comment:
I'd usually put this kind of data preparation in the mapStateToProp : 

```js
const mapStateToProps = state, props => {
  const programCompleteTrainings = props.completeTrainings
    .filter(({ training, program }) => program.id === props.program.id)
    .map(({ training, program }) => training);
  const completeTrainingsCount = programCompleteTrainings.length;
  const firstTodoTraining = props.program.trainings.find(training => !programCompleteTrainings.includes(training));
  return {
    firstTodoTraining, completeTrainingsCount
  };
};
```

But I don't really have a good argument as to why.